Can't compile Starrocks with Clang+gcc13 due to Thrift generated code

<!-- (At least include the following, feel free to add if you have more content) -->

## Refactoring request

Clang doesn't accept the Thrift generated C++ classes as valid, but works fine on GCC.

Errors:

```
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.cpp:35:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.h:38:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService.h:10:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TDispatchProcessor.h:22:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TProcessor.h:24:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/protocol/TProtocol.h:46:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/transport/TTransport.h:23:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/Thrift.h:41:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/vector:66:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:993:50: error: arithmetic on a pointer to an incomplete type 'starrocks::TLoadInfo'
  993 |       { return size_type(this->_M_impl._M_finish - this->_M_impl._M_start); }
      |                          ~~~~~~~~~~~~~~~~~~~~~~~ ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:2041:19: note: in instantiation of member function 'std::vector<starrocks::TLoadInfo>::size' requested here
 2041 |     { return (__x.size() == __y.size()
      |                   ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:3454:39: note: in instantiation of function template specialization 'std::operator==<starrocks::TLoadInfo, std::allocator<starrocks::TLoadInfo>>' requested here
 3454 |     else if (__isset.loads && !(loads == rhs.loads))
      |                                       ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:215:7: note: forward declaration of 'starrocks::TLoadInfo'
  215 | class TLoadInfo;
      |       ^
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.cpp:35:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.h:38:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService.h:10:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TDispatchProcessor.h:22:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TProcessor.h:24:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/protocol/TProtocol.h:46:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/transport/TTransport.h:23:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/Thrift.h:41:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/vector:66:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:370:35: error: arithmetic on a pointer to an incomplete type 'starrocks::TTrackingLoadInfo'
  370 |                       _M_impl._M_end_of_storage - _M_impl._M_start);
      |                       ~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:531:7: note: in instantiation of member function 'std::_Vector_base<starrocks::TTrackingLoadInfo, std::allocator<starrocks::TTrackingLoadInfo>>::~_Vector_base' requested here
  531 |       vector() = default;
      |       ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:3484:3: note: in defaulted default constructor for 'std::vector<starrocks::TTrackingLoadInfo>' first required here
 3484 |   TGetTrackingLoadsResult() {
      |   ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:213:7: note: forward declaration of 'starrocks::TTrackingLoadInfo'
  213 | class TTrackingLoadInfo;
      |       ^
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.cpp:35:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.h:37:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/status.h:20:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/string:42:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/char_traits.h:50:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/type_traits:979:21: error: static assertion failed due to requirement 'std::__is_complete_or_unbounded(std::__type_identity<starrocks::TTrackingLoadInfo>{})': template argument must be a complete class or an unbounded array
  979 |       static_assert(std::__is_complete_or_unbounded(__type_identity<_Tp>{}),
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_construct.h:188:21: note: in instantiation of template class 'std::is_destructible<starrocks::TTrackingLoadInfo>' requested here
  188 |       static_assert(is_destructible<_Value_type>::value,
      |                     ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/alloc_traits.h:948:12: note: in instantiation of function template specialization 'std::_Destroy<starrocks::TTrackingLoadInfo *>' requested here
  948 |       std::_Destroy(__first, __last);
      |            ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:735:7: note: in instantiation of function template specialization 'std::_Destroy<starrocks::TTrackingLoadInfo *, starrocks::TTrackingLoadInfo>' requested here
  735 |         std::_Destroy(this->_M_impl._M_start, this->_M_impl._M_finish,
      |              ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:3484:3: note: in instantiation of member function 'std::vector<starrocks::TTrackingLoadInfo>::~vector' requested here
 3484 |   TGetTrackingLoadsResult() {
      |   ^
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.cpp:35:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/agent/utils.h:38:
In file included from /home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService.h:10:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TDispatchProcessor.h:22:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/TProcessor.h:24:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/protocol/TProtocol.h:46:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/transport/TTransport.h:23:
In file included from /home/samrose/workplace/contrib/starrocks/thirdparty/installed/include/thrift/Thrift.h:41:
In file included from /usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/vector:66:
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:993:50: error: arithmetic on a pointer to an incomplete type 'starrocks::TTrackingLoadInfo'
  993 |       { return size_type(this->_M_impl._M_finish - this->_M_impl._M_start); }
      |                          ~~~~~~~~~~~~~~~~~~~~~~~ ^
/usr/bin/../lib/gcc/x86_64-linux-gnu/13/../../../../include/c++/13/bits/stl_vector.h:2041:19: note: in instantiation of member function 'std::vector<starrocks::TTrackingLoadInfo>::size' requested here
 2041 |     { return (__x.size() == __y.size()
      |                   ^
/home/samrose/workplace/contrib/starrocks/be/src/gen_cpp/build/gen_cpp/FrontendService_types.h:3498:55: note: in instantiation of function template specialization 'std::operator==<starrocks::TTrackingLoadInfo, std::allocator<starrocks::TTrackingLoadInfo>>' requested here
 3498 |     else if (__isset.trackingLoads && !(trackingLoads == rhs.trackingLoads))
      |                                                       ^
```

**Is your refactoring request related to a problem? Please describe.**
<!-- A clear and concise description of what the problem is. Ex. I'm having problems reading the code/add a feature because [...] -->

**Describe the solution you'd like**
<!-- A clear and concise description of the refactoring you want to. -->

I haven't really looked into how to fix this, but worth creating an issue, it would be good to compile the Starrocks project with Clang.

**Describe alternatives you've considered**
<!-- A clear and concise description of any alternative solutions or refactoring method you've considered. -->

**Additional context**
<!-- Add any other context or screenshots about the refactoring request here. -->

Relevant links:

* https://github.com/llvm/llvm-project/issues/59292
* https://github.com/llvm/llvm-project/issues/62166

--- a/be/CMakeLists.txt
+++ b/be/CMakeLists.txt
@@ -620,6 +620,10 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
         # ignore warning from apache-orc
         set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wno-unsafe-buffer-usage")
     endif ()
+    if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "18.0.0")
+        # ignore warning from apache-orc
+        set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} -Wno-switch-default")
+    endif ()
 else ()
     set(CXX_GCC_FLAGS "${CXX_GCC_FLAGS} -fcoroutines")
 endif()

--- a/be/src/exec/es/es_scan_reader.cpp
+++ b/be/src/exec/es/es_scan_reader.cpp
@@ -54,10 +54,8 @@ const std::string SOURCE_SCROLL_SEARCH_FILTER_PATH =
 const std::string DOCVALUE_SCROLL_SEARCH_FILTER_PATH =
         "filter_path=_scroll_id,hits.total,hits.hits._score,hits.hits.fields";
 
-const std::string REQUEST_SCROLL_PATH = "_scroll";
 const std::string REQUEST_PREFERENCE_PREFIX = "&preference=_shards:";
 const std::string REQUEST_SEARCH_SCROLL_PATH = "/_search/scroll";
-const std::string REQUEST_SEPARATOR = "/";
 
 ESScanReader::ESScanReader(const std::string& target, const std::map<std::string, std::string>& props,
                            bool doc_value_mode)

--- a/be/src/exec/tablet_info.cpp
+++ b/be/src/exec/tablet_info.cpp
@@ -24,7 +24,6 @@
 
 namespace starrocks {
 
-static const std::string LOAD_OP_COLUMN = "__op";
 // NOTE: This value should keep the same with the value in FE's `STARROCKS_DEFAULT_PARTITION_VALUE` constant.
 static const std::string STARROCKS_DEFAULT_PARTITION_VALUE = "__STARROCKS_DEFAULT_PARTITION__";
 

--- a/be/src/exprs/agg/percentile_cont.h
+++ b/be/src/exprs/agg/percentile_cont.h
@@ -433,8 +433,8 @@ class PercentileContAggregateFunction final : public PercentileContDiscAggregate
             goal = 0;
         }
 
-        InputCppType junior_elm;
-        InputCppType senior_elm;
+        InputCppType junior_elm{};
+        InputCppType senior_elm{};
 
         if (reverse) {
             kWayMergeSort<LT, InputCppType, true>(grid, b, ls, mp, goal, k, junior_elm, senior_elm);

--- a/be/src/http/action/checksum_action.cpp
+++ b/be/src/http/action/checksum_action.cpp
@@ -52,7 +52,6 @@ const std::string TABLET_ID = "tablet_id";
 // do not use name "VERSION",
 // or will be conflict with "VERSION" in thrift/config.h
 const std::string TABLET_VERSION = "version";
-const std::string SCHEMA_HASH = "schema_hash";
 
 void ChecksumAction::handle(HttpRequest* req) {
     LOG(INFO) << "accept one request " << req->debug_string();

--- a/be/src/http/action/datacache_action.cpp
+++ b/be/src/http/action/datacache_action.cpp
@@ -34,7 +34,6 @@ const static std::string HEADER_JSON = "application/json";
 const static std::string ACTION_KEY = "action";
 const static std::string ACTION_STAT = "stat";
 const static std::string ACTION_APP_STAT = "app_stat";
-const static std::string ACTION_INVALIDATE_ALL = "invalidate_all";
 
 std::string cache_status_str(const DataCacheStatus& status) {
     std::string str_status;

--- a/be/src/http/download_action.cpp
+++ b/be/src/http/download_action.cpp
@@ -50,8 +50,6 @@
 namespace starrocks {
 
 const std::string FILE_PARAMETER = "file";
-const std::string DB_PARAMETER = "db";
-const std::string LABEL_PARAMETER = "label";
 const std::string TOKEN_PARAMETER = "token";
 
 DownloadAction::DownloadAction(ExecEnv* exec_env, const std::vector<std::string>& allow_dirs)

--- a/be/src/http/http_client.h
+++ b/be/src/http/http_client.h
@@ -92,8 +92,10 @@ public:
     }
     // used to get content length
     int64_t get_content_length() const {
-        double cl = 0.0f;
-        curl_easy_getinfo(_curl, CURLINFO_CONTENT_LENGTH_DOWNLOAD, &cl);
+        // CURLINFO_CONTENT_LENGTH_DOWNLOAD is deprecated since v7.55.0
+        // https://curl.se/libcurl/c/CURLINFO_CONTENT_LENGTH_DOWNLOAD.html
+        curl_off_t cl = 0;
+        curl_easy_getinfo(_curl, CURLINFO_CONTENT_LENGTH_DOWNLOAD_T, &cl);
         return cl;
     }
 

--- a/be/src/runtime/load_path_mgr.cpp
+++ b/be/src/runtime/load_path_mgr.cpp
@@ -146,7 +146,6 @@ void LoadPathMgr::get_load_data_path(std::vector<std::string>* data_paths) {
 }
 
 const std::string ERROR_FILE_NAME = "error_log";
-const std::string REJECTED_RECORD_FILE_NAME = "rejected_record";
 
 Status LoadPathMgr::get_load_error_file_name(const TUniqueId& fragment_instance_id, std::string* error_path) {
     std::stringstream ss;

--- a/be/src/storage/persistent_index.cpp
+++ b/be/src/storage/persistent_index.cpp
@@ -3778,7 +3778,7 @@ void PersistentIndex::_get_l2_stat(const std::vector<std::unique_ptr<ImmutableIn
 
                     auto iter = usage_and_size_stat.find(key_size);
                     if (iter == usage_and_size_stat.end()) {
-                        usage_and_size_stat.insert({key_size, {usage, size}});
+                        usage_and_size_stat.insert({static_cast<uint32_t>(key_size), {usage, size}});
                     } else {
                         iter->second.first += usage;
                         iter->second.second += size;

--- a/be/src/storage/segment_flush_executor.cpp
+++ b/be/src/storage/segment_flush_executor.cpp
@@ -170,7 +170,7 @@ Status SegmentFlushToken::submit(DeltaWriter* writer, brpc::Controller* cntl,
     }
 
     auto task = std::make_shared<SegmentFlushTask>(this, writer, cntl, request, response, done);
-    auto submit_st = _flush_token->submit(std::move(task));
+    auto submit_st = _flush_token->submit(task);
     if (submit_st.ok()) {
         closure_guard.release();
     } else {

--- a/be/src/util/c_string.h
+++ b/be/src/util/c_string.h
@@ -15,6 +15,7 @@
 #pragma once
 
 #include <cassert>
+#include <cstdint>
 #include <cstring>
 #include <iostream>
 #include <string_view>

--- a/be/test/exprs/like_test.cpp
+++ b/be/test/exprs/like_test.cpp
@@ -664,7 +664,7 @@ TEST_F(LikeTest, splitLikePatternIntoNgramSet) {
     // pattern contains special characters
     std::string pattern = "abc%_abccc\\%e\\\\\\\\";
     std::vector<std::string> ngram_set;
-    NgramBloomFilterReaderOptions options(4, false);
+    NgramBloomFilterReaderOptions options{4, false};
     VectorizedFunctionCallExpr::split_like_string_to_ngram(pattern, options, ngram_set);
     ASSERT_EQ(6, ngram_set.size());
     ASSERT_EQ("abcc", ngram_set[0]);

--- a/be/test/storage/rowset/rle_page_test.cpp
+++ b/be/test/storage/rowset/rle_page_test.cpp
@@ -115,7 +115,7 @@ public:
             uint32_t seek_off = random() % size;
             ASSERT_TRUE(rle_page_decoder.seek_to_position_in_page(seek_off).ok());
             EXPECT_EQ((int32_t)(seek_off), rle_page_decoder.current_index());
-            CppType ret;
+            CppType ret{};
             copy_one<Type, RlePageDecoder<Type>>(&rle_page_decoder, &ret);
             EXPECT_EQ(values[seek_off], ret);
         }

--- a/be/test/util/bit_stream_utils_test.cpp
+++ b/be/test/util/bit_stream_utils_test.cpp
@@ -174,7 +174,7 @@ TEST(TestBitStreamUtil, TestMixed) {
             EXPECT_EQ(val, parity);
             parity = !parity;
         } else {
-            int val;
+            int val = 0;
             result = reader.GetValue(10, &val);
             EXPECT_EQ(val, i);
         }

--- a/gensrc/thrift/AgentService.thrift
+++ b/gensrc/thrift/AgentService.thrift
@@ -144,6 +144,11 @@ enum TAlterJobType {
     DECOMMISSION_BACKEND = 2
 }
 
+struct TAlterMaterializedViewParam {
+    1: required string column_name
+    2: optional string origin_column_name
+    3: optional Exprs.TExpr mv_expr
+}
 
 // This v2 request will replace the old TAlterTabletReq.
 // TAlterTabletReq should be deprecated after new alter job process merged.
@@ -167,12 +172,6 @@ struct TAlterTabletReqV2 {
     16: optional Descriptors.TDescriptorTable desc_tbl
     17: optional Exprs.TExpr where_expr
     18: optional list<string> base_table_column_names 
-}
-
-struct TAlterMaterializedViewParam {
-    1: required string column_name
-    2: optional string origin_column_name
-    3: optional Exprs.TExpr mv_expr
 }
 
 struct TClusterInfo {

--- a/gensrc/thrift/Descriptors.thrift
+++ b/gensrc/thrift/Descriptors.thrift
@@ -467,10 +467,6 @@ struct TTableFunctionTable {
     10: optional bool parquet_use_legacy_encoding
 }
 
-struct TIcebergSchema {
-    1: optional list<TIcebergSchemaField> fields
-}
-
 struct TIcebergSchemaField {
     // Refer to field id in iceberg schema
     1: optional i32 field_id
@@ -483,6 +479,10 @@ struct TIcebergSchemaField {
 
     // Children fields for struct, map and list(array)
     100: optional list<TIcebergSchemaField> children
+}
+
+struct TIcebergSchema {
+    1: optional list<TIcebergSchemaField> fields
 }
 
 struct TPartitionMap {

--- a/gensrc/thrift/FrontendService.thrift
+++ b/gensrc/thrift/FrontendService.thrift
@@ -116,17 +116,17 @@ struct TShowVariableRequest {
     2: required Types.TVarType varType
 }
 
-// Results of a call to describeTable()
-struct TShowVariableResult {
-    1: required map<string, string> variables
-    2: optional list<TVerboseVariableRecord> verbose_variables
-}
-
 struct TVerboseVariableRecord {
     1: optional string variable_name
     2: optional string value
     3: optional string default_value
     4: optional bool is_changed
+}
+
+// Results of a call to describeTable()
+struct TShowVariableResult {
+    1: required map<string, string> variables
+    2: optional list<TVerboseVariableRecord> verbose_variables
 }
 
 // Valid table file formats
@@ -532,20 +532,16 @@ struct TGetLoadsParams {
     5: optional string load_type
 }
 
-struct TGetLoadsResult {
-    1: optional list<TLoadInfo> loads
-}
-
-struct TGetTrackingLoadsResult {
-    1: optional list<TTrackingLoadInfo> trackingLoads;
-}
-
 struct TTrackingLoadInfo {
     1: optional i64 job_id
     2: optional string label
     3: optional string db
     4: optional list<string> urls
     5: optional string load_type
+}
+
+struct TGetTrackingLoadsResult {
+    1: optional list<TTrackingLoadInfo> trackingLoads;
 }
 
 struct TLoadInfo {
@@ -584,8 +580,8 @@ struct TLoadInfo {
     33: optional i64 num_scan_bytes
 }
 
-struct TGetRoutineLoadJobsResult {
-    1: optional list<TRoutineLoadJobInfo> loads
+struct TGetLoadsResult {
+    1: optional list<TLoadInfo> loads
 }
 
 struct TRoutineLoadJobInfo {
@@ -610,8 +606,8 @@ struct TRoutineLoadJobInfo {
     19: optional string other_msg
 }
 
-struct TGetStreamLoadsResult {
-    1: optional list<TStreamLoadInfo> loads
+struct TGetRoutineLoadJobsResult {
+    1: optional list<TRoutineLoadJobInfo> loads
 }
 
 struct TStreamLoadInfo {
@@ -643,6 +639,10 @@ struct TStreamLoadInfo {
 
 }
 
+struct TGetStreamLoadsResult {
+    1: optional list<TStreamLoadInfo> loads
+}
+
 // getTableNames returns a list of unqualified table names
 struct TGetTablesResult {
   1: list<string> tables
@@ -666,10 +666,6 @@ struct TReportAuditStatisticsResult {
 // Service Protocol Details
 enum FrontendServiceVersion {
   V1
-}
-
-struct TBatchReportExecStatusParams {
-  1: required list<TReportExecStatusParams> params_list
 }
 
 // The results of an INSERT query, sent to the coordinator as part of
@@ -743,6 +739,10 @@ struct TReportExecStatusParams {
   29: optional DataCache.TLoadDataCacheMetrics load_datacache_metrics
 }
 
+struct TBatchReportExecStatusParams {
+  1: required list<TReportExecStatusParams> params_list
+}
+
 struct TReportFragmentFinishParams {
     1: optional Types.TUniqueId query_id
     2: optional Types.TUniqueId fragment_instance_id
@@ -751,6 +751,12 @@ struct TReportFragmentFinishParams {
 
 struct TReportFragmentFinishResponse {
     1: optional Status.TStatus status
+}
+
+struct TAuditStatisticsItem {
+    1: optional i64 scan_rows
+    2: optional i64 scan_bytes
+    3: optional i64 table_id
 }
 
 struct TAuditStatistics {
@@ -767,12 +773,6 @@ struct TReportAuditStatisticsParams {
     1: optional Types.TUniqueId query_id
     2: optional Types.TUniqueId fragment_instance_id
     3: optional TAuditStatistics audit_statistics
-}
-
-struct TAuditStatisticsItem {
-    1: optional i64 scan_rows
-    2: optional i64 scan_bytes
-    3: optional i64 table_id
 }
 
 struct TFeResult {
@@ -1418,10 +1418,6 @@ struct TGetTablesConfigRequest {
     1: optional TAuthInfo auth_info
 }
 
-struct TGetTablesConfigResponse {
-    1: optional list<TTableConfigInfo> tables_config_infos
-}
-
 struct TTableConfigInfo {
     1: optional string table_schema
     2: optional string table_name
@@ -1437,12 +1433,12 @@ struct TTableConfigInfo {
     12: optional i64 table_id
 }
 
-struct TGetPartitionsMetaRequest {
-    1: optional TAuthInfo auth_info
+struct TGetTablesConfigResponse {
+    1: optional list<TTableConfigInfo> tables_config_infos
 }
 
-struct TGetPartitionsMetaResponse {
-    1: optional list<TPartitionMetaInfo> partitions_meta_infos
+struct TGetPartitionsMetaRequest {
+    1: optional TAuthInfo auth_info
 }
 
 struct TPartitionMetaInfo {
@@ -1476,9 +1472,39 @@ struct TPartitionMetaInfo {
     28: optional Types.TTxnType version_txn_type = Types.TTxnType.TXN_NORMAL
 }
 
+struct TGetPartitionsMetaResponse {
+    1: optional list<TPartitionMetaInfo> partitions_meta_infos
+}
+
 struct TGetTablesInfoRequest {
     1: optional TAuthInfo auth_info
     2: optional string table_name;
+}
+
+struct TTableInfo {
+    1: optional string table_catalog
+    2: optional string table_schema
+    3: optional string table_name
+    4: optional string table_type
+    5: optional string engine
+    6: optional i64 version
+    7: optional string row_format
+    8: optional i64 table_rows
+    9: optional i64 avg_row_length
+    10: optional i64 data_length
+    11: optional i64 max_data_length
+    12: optional i64 index_length
+    13: optional i64 data_free
+    14: optional i64 auto_increment
+    15: optional i64 create_time
+    16: optional i64 update_time
+    17: optional i64 check_time
+    18: optional string table_collation
+    19: optional i64 checksum
+    20: optional string create_options
+    21: optional string table_comment
+    22: optional string session_id
+    23: optional i64 table_id
 }
 
 struct TGetTablesInfoResponse {
@@ -1592,32 +1618,6 @@ struct TWarehouseInfo {
 struct TGetWarehousesResponse {
     1: optional Status.TStatus status
     2: optional list<TWarehouseInfo> warehouse_infos;
-}
-
-struct TTableInfo {
-    1: optional string table_catalog
-    2: optional string table_schema
-    3: optional string table_name
-    4: optional string table_type
-    5: optional string engine
-    6: optional i64 version
-    7: optional string row_format
-    8: optional i64 table_rows
-    9: optional i64 avg_row_length
-    10: optional i64 data_length
-    11: optional i64 max_data_length
-    12: optional i64 index_length
-    13: optional i64 data_free
-    14: optional i64 auto_increment
-    15: optional i64 create_time
-    16: optional i64 update_time
-    17: optional i64 check_time
-    18: optional string table_collation
-    19: optional i64 checksum
-    20: optional string create_options
-    21: optional string table_comment
-    22: optional string session_id
-    23: optional i64 table_id
 }
 
 struct TAllocateAutoIncrementIdParam {

--- a/thirdparty/download-thirdparty.sh
+++ b/thirdparty/download-thirdparty.sh
@@ -295,6 +295,14 @@ fi
 cd -
 echo "Finished patching $LZ4_SOURCE"
 
+cd $TP_SOURCE_DIR/$ROCKSDB_SOURCE
+if [ ! -f $PATCHED_MARK ] && [ $ROCKSDB_SOURCE == "rocksdb-6.22.1" ]; then
+    patch -p1 < $TP_PATCH_DIR/rocksdb-6.22.1-metadata-header.patch
+    touch $PATCHED_MARK
+fi
+cd -
+echo "Finished patching $ROCKSDB_SOURCE"
+
 # brpc patch to disable shared library
 cd $TP_SOURCE_DIR/$BRPC_SOURCE
 if [ ! -f $PATCHED_MARK ] && [ $BRPC_SOURCE == "brpc-0.9.5" ]; then

--- a/thirdparty/patches/rocksdb-6.22.1-metadata-header.patch
+++ b/thirdparty/patches/rocksdb-6.22.1-metadata-header.patch
@@ -1,0 +1,84 @@
+diff --git a/include/rocksdb/metadata.h b/include/rocksdb/metadata.h
+index b515c51..ec2fa46 100644
+--- a/include/rocksdb/metadata.h
++++ b/include/rocksdb/metadata.h
+@@ -19,39 +19,6 @@ struct ColumnFamilyMetaData;
+ struct LevelMetaData;
+ struct SstFileMetaData;
+ 
+-// The metadata that describes a column family.
+-struct ColumnFamilyMetaData {
+-  ColumnFamilyMetaData() : size(0), file_count(0), name("") {}
+-  ColumnFamilyMetaData(const std::string& _name, uint64_t _size,
+-                       const std::vector<LevelMetaData>&& _levels)
+-      : size(_size), name(_name), levels(_levels) {}
+-
+-  // The size of this column family in bytes, which is equal to the sum of
+-  // the file size of its "levels".
+-  uint64_t size;
+-  // The number of files in this column family.
+-  size_t file_count;
+-  // The name of the column family.
+-  std::string name;
+-  // The metadata of all levels in this column family.
+-  std::vector<LevelMetaData> levels;
+-};
+-
+-// The metadata that describes a level.
+-struct LevelMetaData {
+-  LevelMetaData(int _level, uint64_t _size,
+-                const std::vector<SstFileMetaData>&& _files)
+-      : level(_level), size(_size), files(_files) {}
+-
+-  // The level which this meta data describes.
+-  const int level;
+-  // The size of this level in bytes, which is equal to the sum of
+-  // the file size of its "files".
+-  const uint64_t size;
+-  // The metadata of all sst files in this level.
+-  const std::vector<SstFileMetaData> files;
+-};
+-
+ // The metadata that describes a SST file.
+ struct SstFileMetaData {
+   SstFileMetaData()
+@@ -153,6 +120,39 @@ struct LiveFileMetaData : SstFileMetaData {
+   LiveFileMetaData() : column_family_name(), level(0) {}
+ };
+ 
++// The metadata that describes a level.
++struct LevelMetaData {
++  LevelMetaData(int _level, uint64_t _size,
++                const std::vector<SstFileMetaData>&& _files)
++      : level(_level), size(_size), files(_files) {}
++
++  // The level which this meta data describes.
++  const int level;
++  // The size of this level in bytes, which is equal to the sum of
++  // the file size of its "files".
++  const uint64_t size;
++  // The metadata of all sst files in this level.
++  const std::vector<SstFileMetaData> files;
++};
++
++// The metadata that describes a column family.
++struct ColumnFamilyMetaData {
++  ColumnFamilyMetaData() : size(0), file_count(0), name("") {}
++  ColumnFamilyMetaData(const std::string& _name, uint64_t _size,
++                       const std::vector<LevelMetaData>&& _levels)
++      : size(_size), name(_name), levels(_levels) {}
++
++  // The size of this column family in bytes, which is equal to the sum of
++  // the file size of its "levels".
++  uint64_t size;
++  // The number of files in this column family.
++  size_t file_count;
++  // The name of the column family.
++  std::string name;
++  // The metadata of all levels in this column family.
++  std::vector<LevelMetaData> levels;
++};
++
+ // Metadata returned as output from ExportColumnFamily() and used as input to
+ // CreateColumnFamiliesWithImport().
+ struct ExportImportFilesMetaData {

